src/frontend/bibletime.cpp

```diff
@@ -332,3 +335,39 @@ void BibleTime::openSearchDialog(BtConstModuleList modules,
 }
 
 void BibleTime::openFindWidget() { m_findWidget->showAndSelect(); }
+
+#ifdef BUILD_TEXT_TO_SPEECH
+void BibleTime::speakText(QString const & text) {
+    if (!m_textToSpeech)
+        m_textToSpeech = createTextToSpeechInstance();
+
+    m_textToSpeech->say(text);
+}
+
+std::unique_ptr<QTextToSpeech> BibleTime::createTextToSpeechInstance() {
+    std::unique_ptr<QTextToSpeech> tts;
+
+    // restore settings from config
+    QString const configuredEngine = btConfig().value<QString>(QStringLiteral("GUI/ttsEngine"));
+    if (QTextToSpeech::availableEngines().contains(configuredEngine)) {
+        tts = std::make_unique<QTextToSpeech>(configuredEngine);
+    } else {
+        tts = std::make_unique<QTextToSpeech>();
+    }
+
+    QLocale const configuredLocale = btConfig().value<QLocale>(QStringLiteral("GUI/ttsLocale"));
+    if (tts->availableLocales().contains(configuredLocale))
+        tts->setLocale(configuredLocale);
+
+    QString const configuredVoice = btConfig().value<QString>(QStringLiteral("GUI/ttsVoice"));
+    QList<QVoice> const voices = tts->availableVoices();
+    for (auto const & voice : voices) {
```

Comment:
It would be more brief to use `for (auto const & voice : tts->availableVoices()) {` as one would not need to spell out the type (or `auto`).

Reply:
Done